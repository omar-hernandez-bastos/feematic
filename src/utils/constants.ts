export enum ActionsPad {
  CLEAR = 'ðŸš«',
  DECIMAL = '.'
}
export const pads = [
  { text: 'ðŸ‡¨ðŸ‡´', className: '', color: 'cyan-200' },
  { text: 'ðŸ‡ºðŸ‡¸', className: '', color: 'gray-100' },
  { text: 'ðŸ‡»ðŸ‡ª', className: 'row-span-2' },
  { text: '7', className: '' },
  { text: '8', className: '' },
  { text: '9', className: '' },
  { text: '4', className: '' },
  { text: '5', className: '' },
  { text: '6', className: '' },
  { text: '1', className: '' },
  { text: '2', className: '' },
  { text: '3', className: '' },
  { text: ActionsPad.CLEAR, className: 'row-span-2' },
  { text: '0', className: 'col-span-2' },
  { text: ActionsPad.DECIMAL, className: '' }
]

export type CURRENCIES = 'ðŸ‡¨ðŸ‡´' | 'ðŸ‡ºðŸ‡¸' | 'ðŸ‡»ðŸ‡ª'
const currencies = {
  COP: 'ðŸ‡¨ðŸ‡´',
  USD: 'ðŸ‡ºðŸ‡¸',
  VES: 'ðŸ‡»ðŸ‡ª'
}
export const CURRENCIES_ARRAY = [currencies.USD, currencies.COP, currencies.VES]
export { currencies }
