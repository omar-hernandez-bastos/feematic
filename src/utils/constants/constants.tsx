export enum ActionsPad {
  CLEAR = 'AC',
  C = 'C',
  DECIMAL = '.',
  CREDIT_CARD = 'ðŸ’³'
}
export const pads = [
  { id: 1, text: 'ðŸ‡¨ðŸ‡´', className: 'col-span-2', color: 'cyan-200' },
  { id: 2, text: 'ðŸ‡ºðŸ‡¸', className: 'col-span-2', color: 'gray-100' },
  { id: 4, text: '7', className: '' },
  { id: 5, text: '8', className: '' },
  { id: 6, text: '9', className: '' },
  { id: 3, text: 'ðŸ‡»ðŸ‡ª', className: '' },
  { id: 7, text: '4', className: '' },
  { id: 8, text: '5', className: '' },
  { id: 9, text: '6', className: '' },
  { id: 3, text: 'ðŸ’³', className: '' },
  { id: 10, text: '1', className: '' },
  { id: 11, text: '2', className: '' },
  { id: 12, text: '3', className: '' },
  { id: 13, text: ActionsPad.C, className: '' },
  { id: 14, text: '0', className: 'col-span-2' },
  { id: 15, text: ActionsPad.DECIMAL, className: '' },
  { id: 16, text: ActionsPad.CLEAR, className: '' }
]

export type CURRENCIES = 'ðŸ‡¨ðŸ‡´' | 'ðŸ‡ºðŸ‡¸' | 'ðŸ‡»ðŸ‡ª'
const currencies = {
  COP: 'ðŸ‡¨ðŸ‡´',
  USD: 'ðŸ‡ºðŸ‡¸',
  VES: 'ðŸ‡»ðŸ‡ª'
}
export const CURRENCIES_ARRAY = [currencies.USD, currencies.COP, currencies.VES]
export { currencies }

export const COP_DEFAULT = 3800
export const VES_DEFAULT = 4.5
