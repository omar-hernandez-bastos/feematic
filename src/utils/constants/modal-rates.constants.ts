const CUSTOM_VES = 'customVes'
const CUSTOM_COP = 'customCop'
const CUSTOM_RATES = 'customRates'
export { CUSTOM_VES, CUSTOM_COP, CUSTOM_RATES }
