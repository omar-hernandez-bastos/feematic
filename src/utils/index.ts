export function classNames(...classes: unknown[]): string {
  return classes.filter(Boolean).join(' ')
}

export function formatMoney(value: number): string {
  const formatter = new Intl.NumberFormat('es-CO', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2
  })
  return formatter.format(value)
}
