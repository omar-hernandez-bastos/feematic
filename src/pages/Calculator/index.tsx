import Pad from 'components/base/Pad'
import Operation from 'components/Operation'
import Result from 'components/Result'
import React, { useState } from 'react'
import { currencies, CURRENCIES_ARRAY, pads } from 'utils/constants'

export interface Amount {
  value: string
  currency: string
}

const Calculator = () => {
  const [amount, setAmount] = useState<Amount>({
    value: '',
    currency: currencies.USD
  })
  const [result, setResult] = useState(0)
  const onPressPad = (value: string) => {
    if (CURRENCIES_ARRAY.includes(value))
      return setAmount({
        ...amount,
        currency: value
      })
    if (value === '‚ùå')
      return setAmount({ value: '', currency: currencies.USD })
    if (value === '.') {
      return amount?.value.includes('.')
        ? null
        : setAmount({
            value: `${amount?.value}${value}`,
            currency: amount?.currency
          })
    }
    return setAmount({
      value: `${amount?.value}${value}`,
      currency: amount?.currency
    })
  }

  return (
    <div className="h-full w-full md:h-[80vh] md:w-96 md:rounded-3xl bg-[#303136] md:shadow-2xl md:shadow-cyan-500/50">
      <div className="p-8 flex flex-col justify-between w-full h-full">
        <div className="w-full h-1/3 flex flex-col items-end justify-center">
          <Operation amount={amount}></Operation>
          <Result amount={amount}></Result>
        </div>
        <div className="w-full h-2/3 grid grid-cols-4 gap-3">
          {pads.map(({ text, className, color }) => (
            <Pad
              className={className}
              onClick={() => onPressPad(text)}
              text={text}
              color={color}
              isActive={amount.currency === text}
            ></Pad>
          ))}
        </div>
      </div>
    </div>
  )
}

export default Calculator
