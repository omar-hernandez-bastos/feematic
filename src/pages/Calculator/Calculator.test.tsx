import { fireEvent, render } from '@testing-library/react'
import { ActionsPad, currencies } from 'utils/constants'
import Calculator from '.'

describe('Calculator Test', () => {
  it('Calculator Component should render correctly', () => {
    const CalculatorProps = {
      cop: 3800,
      ves: 4.5
    }
    const { container } = render(<Calculator {...CalculatorProps} />)
    expect(container).toMatchSnapshot()
  })
  it('Press Pad', () => {
    const CalculatorProps = {
      cop: 3800,
      ves: 4.5
    }
    const { container, getByTestId } = render(
      <Calculator {...CalculatorProps} />
    )
    fireEvent.click(getByTestId(currencies.USD))
    fireEvent.click(getByTestId(ActionsPad.CLEAR))
    fireEvent.click(getByTestId('1'))
    fireEvent.click(getByTestId(ActionsPad.DECIMAL))
    fireEvent.click(getByTestId('0'))
    expect(container).toMatchSnapshot()
  })
})
