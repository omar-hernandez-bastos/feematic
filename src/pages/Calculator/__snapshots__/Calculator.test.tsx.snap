// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Test Calculator Component should render correctly 1`] = `
<div>
  <div
    class="overflow-y-auto relative w-screen md:w-96 h-screen md:h-[90vh] bg-[#303136] md:rounded-3xl md:shadow-2xl md:shadow-cyan-500/50"
  >
    <div
      class="absolute top-2 left-2"
    >
      <button
        class="py-2 px-4 text-sm font-medium text-white hover:text-cyan-500"
        type="button"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.49 3.17c-.38-1.56-2.6-1.56-2.98 0a1.532 1.532 0 01-2.286.948c-1.372-.836-2.942.734-2.106 2.106.54.886.061 2.042-.947 2.287-1.561.379-1.561 2.6 0 2.978a1.532 1.532 0 01.947 2.287c-.836 1.372.734 2.942 2.106 2.106a1.532 1.532 0 012.287.947c.379 1.561 2.6 1.561 2.978 0a1.533 1.533 0 012.287-.947c1.372.836 2.942-.734 2.106-2.106a1.533 1.533 0 01.947-2.287c1.561-.379 1.561-2.6 0-2.978a1.532 1.532 0 01-.947-2.287c.836-1.372-.734-2.942-2.106-2.106a1.532 1.532 0 01-2.287-.947zM10 13a3 3 0 100-6 3 3 0 000 6z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex flex-col justify-between p-8 w-full h-auto"
    >
      <div
        class="flex flex-col justify-center items-end w-full h-[40vh]"
      >
        <div
          class="text-5xl font-bold text-right text-white"
        >
          0,00
          ðŸ‡ºðŸ‡¸
        </div>
        <div
          class="flex flex-col my-2 text-3xl font-light text-right text-gray-300"
        >
          <span>
            0,00 ðŸ‡¨ðŸ‡´
          </span>
          <span>
            0,00 ðŸ‡»ðŸ‡ª
          </span>
        </div>
      </div>
      <div
        class="grid grid-cols-4 gap-3 pb-12 w-full h-[50vh]"
      >
        <div
          class="col-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            color="cyan-200"
            data-testid="ðŸ‡¨ðŸ‡´"
          >
            ðŸ‡¨ðŸ‡´
          </button>
        </div>
        <div
          class="col-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center shadow-2xl shadow-cyan-500/50"
            color="gray-100"
            data-testid="ðŸ‡ºðŸ‡¸"
          >
            ðŸ‡ºðŸ‡¸
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="7"
          >
            7
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="8"
          >
            8
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="9"
          >
            9
          </button>
        </div>
        <div
          class="row-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="ðŸ‡»ðŸ‡ª"
          >
            ðŸ‡»ðŸ‡ª
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="4"
          >
            4
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="5"
          >
            5
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="6"
          >
            6
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="1"
          >
            1
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="2"
          >
            2
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="3"
          >
            3
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="C"
          >
            C
          </button>
        </div>
        <div
          class="col-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="0"
          >
            0
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="."
          >
            .
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="AC"
          >
            AC
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Calculator Test Press Pad 1`] = `
<div>
  <div
    class="overflow-y-auto relative w-screen md:w-96 h-screen md:h-[90vh] bg-[#303136] md:rounded-3xl md:shadow-2xl md:shadow-cyan-500/50"
  >
    <div
      class="absolute top-2 left-2"
    >
      <button
        class="py-2 px-4 text-sm font-medium text-white hover:text-cyan-500"
        type="button"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.49 3.17c-.38-1.56-2.6-1.56-2.98 0a1.532 1.532 0 01-2.286.948c-1.372-.836-2.942.734-2.106 2.106.54.886.061 2.042-.947 2.287-1.561.379-1.561 2.6 0 2.978a1.532 1.532 0 01.947 2.287c-.836 1.372.734 2.942 2.106 2.106a1.532 1.532 0 012.287.947c.379 1.561 2.6 1.561 2.978 0a1.533 1.533 0 012.287-.947c1.372.836 2.942-.734 2.106-2.106a1.533 1.533 0 01.947-2.287c1.561-.379 1.561-2.6 0-2.978a1.532 1.532 0 01-.947-2.287c.836-1.372-.734-2.942-2.106-2.106a1.532 1.532 0 01-2.287-.947zM10 13a3 3 0 100-6 3 3 0 000 6z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex flex-col justify-between p-8 w-full h-auto"
    >
      <div
        class="flex flex-col justify-center items-end w-full h-[40vh]"
      >
        <div
          class="text-5xl font-bold text-right text-white"
        >
          1,00
          ðŸ‡ºðŸ‡¸
        </div>
        <div
          class="flex flex-col my-2 text-3xl font-light text-right text-gray-300"
        >
          <span>
            3.800,00 ðŸ‡¨ðŸ‡´
          </span>
          <span>
            4,50 ðŸ‡»ðŸ‡ª
          </span>
        </div>
      </div>
      <div
        class="grid grid-cols-4 gap-3 pb-12 w-full h-[50vh]"
      >
        <div
          class="col-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            color="cyan-200"
            data-testid="ðŸ‡¨ðŸ‡´"
          >
            ðŸ‡¨ðŸ‡´
          </button>
        </div>
        <div
          class="col-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center shadow-2xl shadow-cyan-500/50"
            color="gray-100"
            data-testid="ðŸ‡ºðŸ‡¸"
          >
            ðŸ‡ºðŸ‡¸
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="7"
          >
            7
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="8"
          >
            8
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="9"
          >
            9
          </button>
        </div>
        <div
          class="row-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="ðŸ‡»ðŸ‡ª"
          >
            ðŸ‡»ðŸ‡ª
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="4"
          >
            4
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="5"
          >
            5
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="6"
          >
            6
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="1"
          >
            1
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="2"
          >
            2
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="3"
          >
            3
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="C"
          >
            C
          </button>
        </div>
        <div
          class="col-span-2"
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="0"
          >
            0
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="."
          >
            .
          </button>
        </div>
        <div
          class=""
        >
          <button
            class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
            data-testid="AC"
          >
            AC
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
